[Bug]: Errors in 2024-10-25 test of `main`

### What happened?

See https://github.com/E3SM-Project/zppy/discussions/634#discussioncomment-11056177

Specifically there are at least 3 errors:

1. `cannot stat` errors for `lnd_monthly_mvm` jobs
2. Image check failures are not being displayed correctly
3. It looks like there is an error in `tc_analysis`. This is likely because expected results need to be updated post-https://github.com/E3SM-Project/e3sm_diags/pull/851

What has changed since the 2024-10-18 run?
[zppy](https://github.com/E3SM-Project/zppy/commits/main):
- None

[e3sm_diags](https://github.com/E3SM-Project/e3sm_diags/commits/main):
- Fixing African easterly wave density plots in TC analysis (https://github.com/E3SM-Project/e3sm_diags/pull/851)

### What machine were you running on?

Chrysalis

### Environment

zppy_dev_weekly_20241025

### What command did you run?

```bash
zppy -c tests/integration/generated/test_weekly_comprehensive_v3_chrysalis.cfg
zppy -c tests/integration/generated/test_weekly_comprehensive_v2_chrysalis.cfg
```


### Copy your cfg file

```bash
# See https://github.com/E3SM-Project/zppy/discussions/634#discussioncomment-11056177
```


### What jobs are failing?

_No response_

### What stack trace are you encountering?

_No response_

After updating the `compare_images` code, I'm able to generate the image check failures:
```
cd ~/ez/zppy
conda activate zppy_dev_weekly_20241025
pip install .
python -u -m unittest tests/integration/test_*.py
```
That gives:
```
======================================================================
FAIL: test_comprehensive_v2_images (tests.integration.test_weekly.TestWeekly)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/gpfs/fs1/home/ac.forsyth2/ez/zppy/tests/integration/test_weekly.py", line 36, in test_comprehensive_v2_images
    test_images(self, "comprehensive_v2", V2_CASE_NAME)
  File "/gpfs/fs1/home/ac.forsyth2/ez/zppy/tests/integration/test_weekly.py", line 27, in test_images
    check_mismatched_images(
  File "/gpfs/fs1/home/ac.forsyth2/ez/zppy/tests/integration/utils.py", line 125, in check_mismatched_images
    test.assertEqual(missing_images, [])
AssertionError: Lists differ: ['e3sm_diags/lnd_monthly_mvm_lnd/model_vs_[25440 chars]png'] != []

First list contains 196 additional elements.
First extra element 0:
'e3sm_diags/lnd_monthly_mvm_lnd/model_vs_model_1982-1983/viewer/viewer/e3sm_logo.png'

Diff is 26080 characters long. Set self.maxDiff to None to see it.

======================================================================
FAIL: test_comprehensive_v3_images (tests.integration.test_weekly.TestWeekly)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/gpfs/fs1/home/ac.forsyth2/ez/zppy/tests/integration/test_weekly.py", line 39, in test_comprehensive_v3_images
    test_images(self, "comprehensive_v3", V3_CASE_NAME)
  File "/gpfs/fs1/home/ac.forsyth2/ez/zppy/tests/integration/test_weekly.py", line 27, in test_images
    check_mismatched_images(
  File "/gpfs/fs1/home/ac.forsyth2/ez/zppy/tests/integration/utils.py", line 125, in check_mismatched_images
    test.assertEqual(missing_images, [])
AssertionError: Lists differ: ['e3sm_diags/lnd_monthly_mvm_lnd/model_vs_[43955 chars]png'] != []

First list contains 336 additional elements.
First extra element 0:
'e3sm_diags/lnd_monthly_mvm_lnd/model_vs_model_1987-1988/viewer/viewer/e3sm_logo.png'

Diff is 45015 characters long. Set self.maxDiff to None to see it.

----------------------------------------------------------------------
Ran 5 tests in 1093.283s

FAILED (failures=2)
```

- https://web.lcrc.anl.gov/public/e3sm/diagnostic_output/ac.forsyth2/zppy_weekly_comprehensive_v2_www/test_zppy_weekly_20241025/v2.LR.historical_0201/image_check_failures_comprehensive_v2/e3sm_diags/atm_monthly_180x360_aave/model_vs_obs_1982-1983/tc_analysis/
- https://web.lcrc.anl.gov/public/e3sm/diagnostic_output/ac.forsyth2/zppy_weekly_comprehensive_v2_www/test_zppy_weekly_20241025/v2.LR.historical_0201/image_check_failures_comprehensive_v2/e3sm_diags/atm_monthly_180x360_aave_mvm/model_vs_model_1980-1981/tc_analysis/

These seem like reasonable changes given https://github.com/E3SM-Project/e3sm_diags/pull/851. 

https://web.lcrc.anl.gov/public/e3sm/diagnostic_output/ac.forsyth2/zppy_weekly_comprehensive_v3_www/test_zppy_weekly_20241025/v3.LR.historical_0051/image_check_failures_comprehensive_v3/ is still empty. That is perhaps expected though.

Looking at the bash output of running `test_weekly.py`:
v2:
```
Missing images:
e3sm_diags/lnd_monthly_mvm_lnd/model_vs_model_1982-1983/viewer/viewer/e3sm_logo.png
e3sm_diags/lnd_monthly_mvm_lnd/model_vs_model_1982-1983/lat_lon_land/model_vs_model/v2.LR.historical_0201-QIRRIG_ORIG-MAM-global.png
e3sm_diags/lnd_monthly_mvm_lnd/model_vs_model_1982-1983/lat_lon_land/model_vs_model/v2.LR.historical_0201-FGEV-ANN-global.png
...
```
and
```
Mismatched images:
e3sm_diags/atm_monthly_180x360_aave_mvm/model_vs_model_1980-1981/tc_analysis/aew-density-map.png
e3sm_diags/atm_monthly_180x360_aave_mvm/model_vs_model_1980-1981/tc_analysis/ace-distribution.png
e3sm_diags/atm_monthly_180x360_aave_mvm/model_vs_model_1980-1981/tc_analysis/tc-frequency.png
e3sm_diags/atm_monthly_180x360_aave_mvm/model_vs_model_1980-1981/tc_analysis/tc-intensity.png
e3sm_diags/atm_monthly_180x360_aave_mvm/model_vs_model_1980-1981/tc_analysis/tc-frequency-annual-cycle.png
e3sm_diags/atm_monthly_180x360_aave/model_vs_obs_1982-1983/tc_analysis/aew-density-map.png
e3sm_diags/atm_monthly_180x360_aave/model_vs_obs_1982-1983/tc_analysis/ace-distribution.png
e3sm_diags/atm_monthly_180x360_aave/model_vs_obs_1982-1983/tc_analysis/tc-frequency.png
e3sm_diags/atm_monthly_180x360_aave/model_vs_obs_1982-1983/tc_analysis/tc-intensity.png
e3sm_diags/atm_monthly_180x360_aave/model_vs_obs_1982-1983/tc_analysis/tc-frequency-annual-cycle.png
```
v3: 
```
Missing images:
e3sm_diags/lnd_monthly_mvm_lnd/model_vs_model_1987-1988/viewer/viewer/e3sm_logo.png
e3sm_diags/lnd_monthly_mvm_lnd/model_vs_model_1987-1988/lat_lon_land/model_vs_model/v3.LR.historical_0051-FIRA-DJF-global.png
e3sm_diags/lnd_monthly_mvm_lnd/model_vs_model_1987-1988/lat_lon_land/model_vs_model/v3.LR.historical_0051-QIRRIG_ORIG-DJF-global.png
...
```
All of the `Missing images` are the `mvm_lnd` error. v3 doesn't have `Mismatched images`, so it makes sense the `image_check_failures` directory is empty.

--- a/tests/integration/utils.py
+++ b/tests/integration/utils.py
@@ -62,13 +62,23 @@ def compare_images(
 
             mismatched_images.append(image_name)
 
+            diff_dir_actual_png = os.path.join(
+                diff_dir, "{}_actual.png".format(image_name)
+            )
+            # image_name could contain a number of subdirectories.
+            os.makedirs(os.path.dirname(diff_dir_actual_png), exist_ok=True)
             shutil.copy(
                 path_to_actual_png,
-                os.path.join(diff_dir, "{}_actual.png".format(image_name)),
+                diff_dir_actual_png,
             )
+            diff_dir_expected_png = os.path.join(
+                diff_dir, "{}_expected.png".format(image_name)
+            )
+            # image_name could contain a number of subdirectories.
+            os.makedirs(os.path.dirname(diff_dir_expected_png), exist_ok=True)
             shutil.copy(
                 path_to_expected_png,
-                os.path.join(diff_dir, "{}_expected.png".format(image_name)),
+                diff_dir_expected_png,
             )
             # https://stackoverflow.com/questions/41405632/draw-a-rectangle-and-a-text-in-it-using-pil
             draw = ImageDraw.Draw(diff)

